mypy/test/testdmypy.py

```diff
@@ -0,0 +1,308 @@
+"""Type checker test cases"""
```

Comment:
This started out as a clone of testcheck.py, but there are many small changes.

Reply:
I'd really like to be able to refactor the duplications in the test files in order to eventually clean up the test logic and improve the usability (#3973 being a first step). A comment about the nature of these differences may help with that.

Reply:
Thanks.